magenta/js/music_rnn/src/model.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import * as dl from 'deeplearn';
+import * as magenta from '@magenta/core';
+import * as data from './data';
+import { INoteSequence } from '@magenta/core';
+
+// tslint:disable-next-line:max-line-length
+const CHECKPOINT_URL = "https://storage.googleapis.com/download.magenta.tensorflow.org/models/music_rnn/dljs/basic_rnn/";
+
+const DEFAULT_MIN_NOTE = 48;
+const DEFAULT_MAX_NOTE = 83;
+
+/**
+ * Main MusicVAE model class.
+ *
+ * A MusicVAE is a variational autoencoder made up of an `Encoder` and
+ * `Decoder`, along with a `DataConverter` for converting between `Tensor`
+ * and `NoteSequence` objects for input and output.
+ *
+ * Exposes methods for interpolation and sampling of musical sequences.
+ */
+export class MelodyRnn {
+  checkpointURL: string;
+
+  lstmKernel1: dl.Tensor2D;
+  lstmBias1: dl.Tensor1D;
+  lstmKernel2: dl.Tensor2D;
+  lstmBias2: dl.Tensor1D;
+  lstmFcB: dl.Tensor1D;
+  lstmFcW: dl.Tensor2D;
+
+  initialized: boolean;
+
+  /**
+   * `MusicVAE` constructor.
+   *
+   * @param checkpointURL Path to the checkpoint directory.
+   * @param dataConverter A `DataConverter` object to use for converting between
+   * `NoteSequence` and `Tensor` objects. If not provided, a `converter.json`
+   * file must exist within the checkpoint directory specifying the type and
+   * args for the correct `DataConverter`.
+   */
+  constructor(checkpointURL=CHECKPOINT_URL) {
+    this.checkpointURL = checkpointURL;
+    this.initialized = false;
+  }
+
+  /**
+   * Loads variables from the checkpoint and instantiates the `Encoder` and
+   * `Decoder`.
+   */
+  async initialize() {
+    const reader = new dl.CheckpointLoader(this.checkpointURL);
+    const vars = await reader.getAllVariables();
+
+    // Model variables.
+    // tslint:disable:max-line-length
+    this.lstmKernel1 = vars['RNN/MultiRNNCell/Cell0/BasicLSTMCell/Linear/Matrix'] as dl.Tensor2D;
```

Comment:
This looks to be the old naming format. We should use the new one and update the checkpoints as needed.

Reply:
Done